### ISSUE TITLE
Points drawn with ImmediateGeometry won't render over certain number.

### ISSUE THREAD

**Issue by dodgyville** [godotengine/godot#23444]
**Godot version:**
3.1 master (a332ba497b03655212600e39b584d6d21b38a065) but behaviour not new

**OS/device including version:**
2013 macbook air and linux ubuntu 18.04 

**Issue description:**
I am drawing a point cloud and using ImmediateGeometry with Mesh.PRIMITIVE_POINTS. On my machine it draws 70,000 points with no problems but if I use 80,000 the points are missing (there is a point at (0,0,0). 

Is there a limit to how many points I can draw this way (perhaps hardware related) or is this a bug?

**Steps to reproduce:**
```
	var point_size = 5
	var im = ImmediateGeometry.new()
	add_child(im)
	var m = SpatialMaterial.new()
	m.flags_use_point_size = true
	m.params_point_size = point_size
	m.vertex_color_use_as_albedo = true
	im.set_material_override(m)
	im.clear()
	im.begin(Mesh.PRIMITIVE_POINTS, null)
	im.set_color(Color(0,1,1))
	var s = 20
	for i in 80000: # 70000 draw fine, 80000 none draw or they all draw at 0,0,0 perhaps
		im.add_vertex(Vector3(rand_range(-s, s), rand_range(-s, s), rand_range(-s, s)))
	im.end()

```

**Comment by dodgyville** [comment]
A work-around is to create multiple ImmediateGeometry objects

**Comment by Calinou** [comment]
Try increasing **Rendering > Limits > Buffers > Immediate Buffer Size Kb** in the Project Settings.

**Comment by tom-jk** [comment]
Tested with 3.2.2.stable.
Same behaviour (shows only one point at origin when point count exceeds 74898).
Suggested fix (increasing immediate buffer size) works as expected (double buffer size = double max points).
All appears to be working as intended.

convenience project:
[23444.zip](https://github.com/godotengine/godot/files/4902133/23444.zip)

**Comment by ChristopheLY** [comment]
On the branch master (4.0), 
I can't find `SpatialMaterial` and  say this isn't declared
another thing, there aren't **Immediate Buffer Size Kb** in Project Settings (It's renamed?)

**Comment by Calinou** [comment]
@ChristopheLY SpatialMaterial was renamed to StandardMaterial3D. Also, ImmediateGeometry isn't reimplemented yet in the `master` branch.

**Comment by akien-mga** [comment]
Fixed by #43208.
